packages/email/email.js

```diff
@@ -227,27 +225,66 @@ Email.customTransport = undefined;
  * @param {MailComposer} [options.mailComposer] A [MailComposer](https://nodemailer.com/extras/mailcomposer/#e-mail-message-fields)
+ * @param {Object} [options.stream] Output stream to write email on development environment
  * object representing the message to be sent.  Overrides all other options.
  * You can create a `MailComposer` object via
  * `new EmailInternals.NpmModules.mailcomposer.module`.
  */
-Email.send = function(options) {
-  if (options.mailComposer) {
-    options = options.mailComposer.mail;
-  }
+Email.send = function (options) {
+  // Using Fibers Promise.await
+  Promise.await(Email.sendAsync(options));
+};
+
+/**
+ * @summary Send an email with asyncronous method. Capture  Throws an `Error` on failure to contact mail server
+ * or if mail server returns an error. All fields should match
+ * [RFC5322](http://tools.ietf.org/html/rfc5322) specification.
+ *
+ * If the `MAIL_URL` environment variable is set, actually sends the email.
+ * Otherwise, prints the contents of the email to standard out.
+ *
+ * Note that this package is based on **nodemailer**, so make sure to refer to
+ * [the documentation](http://nodemailer.com/)
+ * when using the `attachments` or `mailComposer` options.
+ *
+ * @locus Server
+ * @return {Promise}
+ * @param {Object} options
+ * @param {String} [options.from] "From:" address (required)
+ * @param {String|String[]} options.to,cc,bcc,replyTo
+ *   "To:", "Cc:", "Bcc:", and "Reply-To:" addresses
+ * @param {String} [options.inReplyTo] Message-ID this message is replying to
+ * @param {String|String[]} [options.references] Array (or space-separated string) of Message-IDs to refer to
+ * @param {String} [options.messageId] Message-ID for this message; otherwise, will be set to a random value
+ * @param {String} [options.subject]  "Subject:" line
+ * @param {String} [options.text|html] Mail body (in plain text and/or HTML)
+ * @param {String} [options.watchHtml] Mail body in HTML specific for Apple Watch
+ * @param {String} [options.icalEvent] iCalendar event attachment
+ * @param {Object} [options.headers] Dictionary of custom headers - e.g. `{ "header name": "header value" }`. To set an object under a header name, use `JSON.stringify` - e.g. `{ "header name": JSON.stringify({ tracking: { level: 'full' } }) }`.
+ * @param {Object[]} [options.attachments] Array of attachment objects, as
+ * described in the [nodemailer documentation](https://nodemailer.com/message/attachments/).
+ * @param {MailComposer} [options.mailComposer] A [MailComposer](https://nodemailer.com/extras/mailcomposer/#e-mail-message-fields)
+ * @param {Object} [options.stream] Output stream to write email on development environment
+ * object representing the message to be sent.  Overrides all other options.
+ * You can create a `MailComposer` object via
+ * `new EmailInternals.NpmModules.mailcomposer.module`.
+ */
+Email.sendAsync = async function (options) {
+  const { stream = process.stdout, ...rest } = options;
```

Comment:
This destructioning doesn't make much sense to me. It should either get `mailComposer` for use on the line bellow or it should be `const stream = options.stream || process.stdout` or `const { stream = process.stdout } = options;`
So that on the next line you just use the full options object (unless that is an issue).
Also stream should be defined just before it is used on line 306 so that you don't needlessly define it here when the function can be returned before it is even used.

Reply:
we are not creating any unnecessary resource, declaring in the top makes it easy to see how it is going to define if no stream property is provided.

Reply:
I changed and removed it from this method, but I preserved it on the devModeSendAsync method because it's required in async tests to work.